src/network/https.js

```diff
@@ -165,6 +240,17 @@ class HttpsWorker {
           // The refresh token is not valid.
           throw new Error(`Unable to refresh expired token: ${error}`);
         }
+        try {
+          // and call the handler for this event if we have one
+          if (typeof (this.events.refresh) === 'function') {
+            await this.events.refresh(this.tokens);
+          }
+        } catch (error) {
+          // what should we do here?
```

Comment:
i think what you did is good. Throwing an error to tell the user the handler ran into an issue is desired.